web/static/js/thesis-editor.js

```diff
@@ -119,11 +173,23 @@ class ThesisEditor extends React.Component {
 
   postToServer (page, contents) {
     Net.put('/thesis/update', {page, contents}).then((resp) => {
-      console.log('SUCCESS')
-      console.log(resp)
+      if (page.slug != window.location.pathname) {
+        window.location = page.slug
```

Comment:
If we've changed the current page slug, or are creating a new page, redirect to the new URL.